[Metrics] Add input/output token and request size metrics (#214)
kubernetes-sigs/gateway-api-inference-extension: pkg/ext-proc/metrics/README.md

* [Exposed Metrics](#exposed-metrics)
* [Scrape Metrics](#scrape-metrics)

## Requirements

> In order to process response metrics, you will need to enable `Buffered` mode for response in `EnvoyExtensionPolicy`:

Reviewer: Pls mention that there is a work to enable both streaming and buffered mode https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/178

Author: updated

* [Exposed Metrics](#exposed-metrics)
* [Scrape Metrics](#scrape-metrics)

## Requirements

> > There is an ongoing work to support both streaming and buffered model: https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/178. Without it, each response for streamed request will process one token for each response, resulting in multiple small increasement to response metrics.

Reviewer: ```suggestion
```

Author: Updated

> In order to process response metrics, you will need to enable streaming, or enable `Buffered` mode for response in `EnvoyExtensionPolicy`:

Reviewer: ```suggestion

NOTE: Response metrics are only supported in non-streaming mode, with the follow up [issue](https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/178) to address streaming mode.

Currently you have 2 options:
- If your use response streaming, simply leave the response body processing mode empty in your `EnvoyExtensionPolicy` (default). You won't get response metrics reporting.
- If you don't use streaming, to enable  response metrics reporting, you can enable  `Buffered` mode for response in `EnvoyExtensionPolicy`.
```

Author: Done.

* [Exposed Metrics](#exposed-metrics)
* [Scrape Metrics](#scrape-metrics)

## Requirements

> NOTE: Response metrics are only supported in non-streaming mode, with the follow up [issue](https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/178) to address streaming mode.

Reviewer 2: ```suggestion
Response metrics are only supported in non-streaming mode, with the follow up [issue](https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/178) to address streaming mode.
```

Currently you have 2 options:

> - If your use response streaming, simply leave the response body processing mode empty in your `EnvoyExtensionPolicy` (default). You won't get response metrics reporting.

Reviewer 2: ```suggestion
- If you use response streaming, simply leave the response body processing mode empty in your `EnvoyExtensionPolicy` (default). You won't get response metrics reporting.
```

> Currently you have 2 options:
> - If you use response streaming, simply leave the response body processing mode empty in your `EnvoyExtensionPolicy` (default). You won't get response metrics reporting.
> 
> - If you don't use streaming, to enable response metrics reporting, you can enable `Buffered` mode for response in `EnvoyExtensionPolicy`.

Reviewer 2: ```suggestion
Currently there are two options:
- If requests don't use response streaming, then you can enable `Buffered` mode for response in `EnvoyExtensionPolicy`, this will buffer the response body at the proxy and forward it to the endpoint picker, which allows the endpoint picker to report response metrics.

- If requests use response streaming, then it is not recommended to enable `Buffered` mode, the response body processing mode should be left empty in the `EnvoyExtensionPolicy` (default). In this case response bodies will not be forwarded to the endpoint picker, and therefore response metrics will not be reported.

```

> - If you don't use streaming, to enable  response metrics reporting, you can enable  `Buffered` mode for response in `EnvoyExtensionPolicy`.

Reviewer 2: ```suggestion
- If you don't use streaming, to enable response metrics reporting, you can enable `Buffered` mode for response in `EnvoyExtensionPolicy`.
```
